src/python-common/ceph/deployment/service_spec.py

```diff
@@ -176,7 +178,7 @@ def __repr__(self):
     def from_json(cls, data):
         hosts = data.get('hosts', [])
         if hosts:
-            data['hosts'] = [HostPlacementSpec.from_json(host) for host in hosts]
+            data['hosts'] = [HostPlacementSpec.parse(host) for host in hosts]
```

Comment:
`to_json()` also needs this change.

Reply:
comming back to this, I don't see the necessity to do that anymore. Do you?

Reply:
As long as `HostPlacementSpec.from_json(HostPlacementSpec(...).to_json())` works, I'm happy!

Reply:
covered in the next batch of tests